evm/hyperliquid-block-helper.js

```diff
@@ -0,0 +1,182 @@
+        -H "User-Agent: Mozilla/5.0 (compatible; Hyperliquid-Block-Helper/1.0)" \
+        -d '${JSON.stringify(payload)}' \
+        --connect-timeout 15 \
+        --max-time 30`;
+
+    const { stdout, stderr } = await execAsync(curlCommand);
+
+    if (stderr && !stderr.includes('curl')) {
+        throw new Error(`curl stderr: ${stderr}`);
+    }
+
+    const result = JSON.parse(stdout);
+    return result;
+}
+
+async function switchBlockSize(privateKey, useBig, network = 'mainnet') {
+    validateParameters({ isNonEmptyString: { privateKey } });
+
+    if (!privateKey.startsWith('0x')) {
+        privateKey = '0x' + privateKey;
+    }
+
+    if (privateKey.length !== 66) {
+        throw new Error(`Invalid private key length: ${privateKey.length}`);
+    }
+
+    const wallet = new Wallet(privateKey);
+    const isMainnet = network.toLowerCase() === 'mainnet';
+
+    const action = { type: 'evmUserModify', usingBigBlocks: useBig };
+    const nonce = Date.now();
+    const signature = await signL1Action(wallet, action, null, nonce, isMainnet);
+    const result = await sendRequest(action, signature, nonce, isMainnet);
+
+    if (result.status === 'ok') {
+        return result;
+    } else {
+        throw new Error(`API error: ${result.response}`);
+    }
+}
+
+async function processCommand(config, chain, options) {
+    const { privateKey, blockSize } = options;
+
+    validateParameters({
+        isNonEmptyString: { privateKey },
+        isNonEmptyString: { blockSize },
+    });
+
+    if (blockSize !== 'big' && blockSize !== 'small') {
+        throw new Error('Block size must be "big" or "small"');
+    }
+
+    // Check if this is a Hyperliquid chain
+    const isHyperliquidChain = chain.name.toLowerCase().includes('hyperliquid') || chain.axelarId.toLowerCase().includes('hyperliquid');
+
+    if (!isHyperliquidChain) {
+        throw new Error(`Chain "${chain.name}" is not supported. This script only works on Hyperliquid chains.`);
+    }
+
+    const isMainnet = options.env === 'mainnet';
+    const network = isMainnet ? 'mainnet' : 'testnet';
+    const useBig = blockSize === 'big';
+
+    printInfo('Block size', blockSize.toUpperCase());
+    printInfo('Network', network);
+
+    try {
+        const result = await switchBlockSize(privateKey, useBig, network);
+        printInfo('Result', result);
+        return result;
+    } catch (error) {
+        if (error.message.includes('does not exist')) {
+            printInfo('Result', 'Account not found, continuing without block size switch');
+            return { status: 'ok', message: 'Account not found, continuing without block size switch' };
+        }
+        if (error.message.includes('curl request failed')) {
+            printInfo('Result', 'Block size switch failed, continuing with deployment');
+            return { status: 'ok', message: 'Block size switch failed, continuing with deployment' };
+        }
+        throw error;
+    }
```

Comment:
parse the API response in the callee and post info there vs parsing thrown exceptions

Reply:
or just print the api response error

Reply:
Yeah, throwing the error is preferable, let the caller decide what to do with it